pylabrobot/liquid_handling/backends/hamilton/STAR_backend.py

```diff
@@ -8663,6 +8663,143 @@ def mm_to_z_drive_increment(value_mm: float) -> int:
   def z_drive_increment_to_mm(value_increments: int) -> float:
     return round(value_increments * STARBackend.z_drive_mm_per_increment, 2)
 
+  async def clld_probe_x_position_using_channel(
+    self,
+    channel_idx: int,  # 0-based indexing of channels!
+    probing_direction: Literal["right", "left"],
+    start_pos_search: Optional[float] = None,  # mm
+    end_pos_search: Optional[float] = None,  # mm
+    move_to_safe_z_before_move_to_start_pos: bool = True,
+    post_detection_dist: float = 2.0,  # mm,
+    tip_bottom_diameter: float = 1.2,  # mm
+    read_timeout=240.0,  # seconds
+  ) -> float:
+    """
+    Probe the x-position of a conductive material using a channel’s capacitive liquid
+    level detection (cLLD) via a lateral X scan.
+
+    The channel is optionally moved to a safe Z, positioned at a start X coordinate,
+    and then scanned in the specified direction using the XL command until cLLD
+    triggers. After detection, the channel retreats inward by `post_detection_dist`.
+
+    The returned value is a first-order geometric estimate of the material boundary
+    position in millimeters, corrected by half the tip bottom diameter assuming
+    cylindrical tip contact.
+
+    Assumptions:
+    - Channels can move between x-coordinates 95 - 800 (STARlet) or 95 - 1415 (STAR).
+    - XL motion is constrained to the interval between start and end positions.
+    - `post_detection_dist` is non-negative and always moves the channel inward.
+
+    Side effects:
+    - Moves the specified channel in X and Z.
+    - Leaves the channel at a retracted X position after detection.
+
+    Returns:
+      float: Estimated x-position of the detected material boundary in millimeters.
+    """
+
+    assert channel_idx in range(
+      self.num_channels
+    ), f"Channel index must be between 0 and {self.num_channels - 1}, is {channel_idx}."
+    assert probing_direction in [
+      "right",
+      "left",
+    ], f"Probing direction must be either 'right' or 'left', is {probing_direction}."
+    assert post_detection_dist >= 0.0, (
+      f"Post-detection distance must be non-negative, is {post_detection_dist} mm."
+      "(always marks a movement away from the detected material)."
+    )
+
+    # TODO: Anti-channel-crash feature -> use self.deck with recursive logic
+    current_x_position = await self.request_x_pos_channel_n(channel_idx)
+    # y_position = await self.request_y_pos_channel_n(channel_idx)
+    current_z_position = await self.request_z_pos_channel_n(channel_idx)
+
+    # Use identified rail number to calculate possible upper limit:
+    # STAR = 95 - 1415 mm, STARlet = 95 - 800mm
+    num_rails = self.extended_conf["xt"]
+    track_width = 22.5  # mm
+    reachable_dist_to_last_rail = 125.0
+
+    max_safe_upper_x_pos = num_rails * track_width + reachable_dist_to_last_rail
+    max_safe_lower_x_pos = 95.0  # unit: mm
+
+    if start_pos_search is None:
+      start_pos_search = current_x_position
+    else:
```

Comment:
given how powerful the x-arm is, I would prefer having the user move to start_pos themselves and only support x-probing from the current position. that means removing the start_pos_search parameter and always starting at current_x_position